course5/homework06/string.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+#include <memory>
+#include <ostream>
+#include <cstdlib>
+
+
+class String
+{
+public:
+    String() = default;
+    String(String &&);
+    String(char* data);
+    String(const char* data);
+    String(const String& data);
+
+    String& operator = (const String& string);
+    String& operator = (String && string);
+    String& operator = (const char* data);
+
+
+    char& operator [](std::size_t index);
+    const char& operator [] (std::size_t index) const;
```

Comment:
Possibly. char is 1 byte, the reference is 8, but I suspect it's all the same on 64 bit machines.